### PR TITLE
BIA-777: changing StudentProgramCohortDim view to add entry level and cohort type descriptions

### PR DESCRIPTION
**Pull request by nsd-daniel**
Adding columns to view query from descriptor table
    - EntryGradeDescriptor.Description AS EntryGradeLevelDescriptor
    - CohortTypeDescriptor.Description AS CohortTypeDescriptor

Adding changes in both 3.1 and 3.2 version
Adding changes in both MSSQL and Postgress in version 3.2
Adding new columns to dictionary test for TeamCity testing routine

### CODE REVIEW

**Comment by DavidJGapCR** [comment]
Please generate a new script, instead of changing an existing one.

**Review by bptillman — CHANGES_REQUESTED** [review]
@nsd-daniel there are a few initial issues I see with this PR that I'd like you to address before I dig into it much further. These items are:

1. Please fill in a description in the PR of what these changes are. The title gives some detail about the change, but it is generally a good idea to fill in a description of a pull request to give a little more detail about the changes that were done
2. You need to confirm the CLA Assistant Agreement before this PR can be approved and merged. There is a link here on the main PR page that should take you to the agreement page and once you've done that, the CLA Assistant warning should go away. You should only have to do this once, since this is your first PR on the AMT repo.
3. It does not appear that your commit is signed. For the AMT project we require that all commits be signed. You can follow the guide here to see about getting everything setup locally for having your commits signed: https://techdocs.ed-fi.org/display/ETKB/Signing+Git+Commits
4. There are tests failing currently on TeamCity for this build. Could you please see why? I believe this is due to you adding columns to the view, but then not updating the tests to verify those new columns exist. Typically when modifying any of the AMT views where you are adding columns will require the tests that check the schema to be updated, and could possible involve updating the data load portion of the tests to add any additional data that is needed. So please take a look and see if one or both of these things need to be done.
5. In the AMT project when you update a view like this, you should be adding a new file to update the view. This file you modified was the initial one that created the view. What you would instead want to do is add a new file into this same MSSQL folder with a name like `0009-View-StudentProgramCohortDim-Alter.sql`. Giving it a name like this is since the scripts get run in a specified order (why there is the 0009), and it helps indicate we are altering the view and not creating it for the first time.
6. There are many views we have that exist in Data Standard 3.2 like the file you modified here, but they also can exist in Data Standard 3.1 and 2.0 of the ODS as well. In this case, for the Equity collection, it does not exist in 2.0, so you would only be needing to make similar changes in the DataStandard3.1 project in addition to these changes in the DataStandard3.2 project.

**Comment by DavidJGapCR** [comment]
In our planning meeting, when we estimated this ticket, we talked about creating a new view. But that is not part of the description of the ticket, so I am not completely sure.

**Comment by nsd-daniel** [comment]
> @nsd-daniel there are a few initial issues I see with this PR that I'd like you to address before I dig into it much further. These items are:
> 
> 1. Please fill in a description in the PR of what these changes are. The title gives some detail about the change, but it is generally a good idea to fill in a description of a pull request to give a little more detail about the changes that were done
> 2. You need to confirm the CLA Assistant Agreement before this PR can be approved and merged. There is a link here on the main PR page that should take you to the agreement page and once you've done that, the CLA Assistant warning should go away. You should only have to do this once, since this is your first PR on the AMT repo.
> 3. It does not appear that your commit is signed. For the AMT project we require that all commits be signed. You can follow the guide here to see about getting everything setup locally for having your commits signed: https://techdocs.ed-fi.org/display/ETKB/Signing+Git+Commits . Please note that since you already made the commit, you will need to follow the part of the guide under Troubleshooting for "Need to Sign Previous Commit(s)".
> 4. There are tests failing currently on TeamCity for this build. Could you please see why? I believe this is due to you adding columns to the view, but then not updating the tests to verify those new columns exist. Typically when modifying any of the AMT views where you are adding columns will require the tests that check the schema to be updated, and could possible involve updating the data load portion of the tests to add any additional data that is needed. So please take a look and see if one or both of these things need to be done.
> 5. In the AMT project when you update a view like this, you should be adding a new file to update the view. This file you modified was the initial one that created the view. What you would instead want to do is add a new file into this same MSSQL folder with a name like `0009-View-StudentProgramCohortDim-Alter.sql`. Giving it a name like this is since the scripts get run in a specified order (why there is the 0009), and it helps indicate we are altering the view and not creating it for the first time.
> 6. There are many views we have that exist in Data Standard 3.2 like the file you modified here, but they also can exist in Data Standard 3.1 and 2.0 of the ODS as well. In this case, for the Equity collection, it does not exist in 2.0, so you would only be needing to make similar changes in the DataStandard3.1 project in addition to these changes in the DataStandard3.2 project.

---------------------------------

All requested changes already made.

File: src/EdFi.AnalyticsMiddleTier.Tests/Classes/StudentProgramCohortDim.cs [hunk 1]
```diff
@@ -14,6 +14,8 @@ public class StudentProgramCohortDim
         public string StudentProgramCohortKey { get; set; }
         public string StudentSchoolProgramKey { get; set; }
         public string StudentSchoolKey { get; set; }
+        public string EntryGradeLevelDescriptor { get; set; }
```
Comment (bptillman) [inline src/EdFi.AnalyticsMiddleTier.Tests/Classes/StudentProgramCohortDim.cs:17]: You correctly updated the should_match_column_dictionary test for these two new columns, but we should also include tests to verify expected values come back for these new columns. Take a look at `When_querying_the_StudentProgramCohortDim_view` and inside there should will see tests like `Then_should_have_StudentProgramCohortKey`, `Then_should_have_CohortDescription`, and `Then_should_have_ProgramName` amongst others. We should have similar tests for these two new fields, so you will need to make a new test case in that file for it, along with creating the needed xml files inside `TestCases\StudentProgramCohortDim` .

**Review by bptillman — CHANGES_REQUESTED** [review]

**Review by bptillman — APPROVED** [review]

**Pull request merged**